From 16de74108b1133b596c21b417ab1809aa0f26f60 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <oliver@onion.private>
Date: Sat, 28 Jun 2014 15:26:23 -0400
Subject: [PATCH 0198/1936] Use check boxes rather than light boxes since the
 muted palette of ntk Cairo theme shows their state poorly.  Tidy the
 positions and box sizes for ADvoice and Config UIs.

---
 src/UI/BankUI.fl     | 18 +++++++++---------
 src/UI/ConfigUI.fl   |  8 ++++----
 src/UI/EnvelopeUI.fl |  2 +-
 src/UI/OscilGenUI.fl | 40 ++++++++++++++++++++--------------------
 4 files changed, 34 insertions(+), 34 deletions(-)

diff --git a/src/UI/BankUI.fl b/src/UI/BankUI.fl
index 70c273a2..08b4702d 100644
--- a/src/UI/BankUI.fl
+++ b/src/UI/BankUI.fl
@@ -152,32 +152,32 @@ class BankUI {open : {public BankProcess_}
       Fl_Group modeselect {
         xywh {5 528 425 42} box ENGRAVED_BOX
       } {
-        Fl_Light_Button writebutton {
+        Fl_Check_Button writebutton {
           label WRITE
           callback {if (o->value()>0.5) mode=2;
 removeselection();}
-          xywh {116 534 99 30} type Radio down_box THIN_DOWN_BOX selection_color 1 labeltype ENGRAVED_LABEL labelfont 1 labelsize 13
+          xywh {116 534 99 30} type Radio box UP_BOX down_box DOWN_BOX labelfont 1 labelsize 13
           code0 {if (bank->locked()) o->deactivate();}
         }
-        Fl_Light_Button readbutton {
-          label READ
+        Fl_Check_Button readbutton {
+          label READ selected
           callback {if (o->value()>0.5) mode=1;
 removeselection();}
-          xywh {11 534 99 30} type Radio down_box THIN_DOWN_BOX selection_color 101 labeltype ENGRAVED_LABEL labelfont 1 labelsize 13
+          xywh {11 534 99 30} type Radio box UP_BOX down_box DOWN_BOX labelfont 1 labelsize 13
           code0 {o->value(1);}
         }
-        Fl_Light_Button clearbutton {
+        Fl_Check_Button clearbutton {
           label CLEAR
           callback {if (o->value()>0.5) mode=3;
 removeselection();}
-          xywh {221 534 99 30} type Radio down_box THIN_DOWN_BOX selection_color 0 labeltype ENGRAVED_LABEL labelfont 1 labelsize 13
+          xywh {221 534 99 30} type Radio box UP_BOX down_box DOWN_BOX labelfont 1 labelsize 13
           code0 {if (bank->locked()) o->deactivate();}
         }
-        Fl_Light_Button swapbutton {
+        Fl_Check_Button swapbutton {
           label SWAP
           callback {if (o->value()>0.5) mode=4;
 removeselection();}
-          xywh {325 534 99 30} type Radio down_box THIN_DOWN_BOX selection_color 227 labeltype ENGRAVED_LABEL labelfont 1 labelsize 13
+          xywh {325 534 99 30} type Radio box UP_BOX down_box DOWN_BOX labelfont 1 labelsize 13
           code0 {if (bank->locked()) o->deactivate();}
         }
       }
diff --git a/src/UI/ConfigUI.fl b/src/UI/ConfigUI.fl
index b81c54a8..c7ec12d5 100644
--- a/src/UI/ConfigUI.fl
+++ b/src/UI/ConfigUI.fl
@@ -112,10 +112,10 @@ config.cfg.SoundBufferSize=strtoul(o->value(),&tmp,10);}
             code1 {snprintf(tmpbuf,100,"%d",config.cfg.SoundBufferSize);o->insert(tmpbuf);}
             code2 {delete []tmpbuf;}
           }
-          Fl_Light_Button {} {
+          Fl_Check_Button {} {
             label {Swap Stereo }
             callback {config.cfg.SwapStereo=(int) o->value();}
-            xywh {20 80 85 20} box THIN_UP_BOX labelsize 10
+            xywh {15 80 95 20} box NO_BOX labelsize 11
             code0 {o->value(config.cfg.SwapStereo);}
           }
           Fl_Choice {} {
@@ -178,7 +178,7 @@ config.cfg.SoundBufferSize=strtoul(o->value(),&tmp,10);}
               label {Dump notes}
               callback {config.cfg.DumpNotesToFile=(int) o->value();
 dump.startnow();//this has effect only if this option was disabled}
-              xywh {20 130 100 20} down_box DOWN_BOX
+              xywh {20 130 110 20} down_box DOWN_BOX
               code0 {o->value(config.cfg.DumpNotesToFile);}
             }
             Fl_Check_Button {} {
@@ -239,7 +239,7 @@ dump.startnow();//this has effect only if this option was disabled}
           Fl_Check_Button {} {
             label {Ignore MIDI Program Change}
             callback {config.cfg.IgnoreProgramChange=(int) o->value();}
-            xywh {10 255 230 20} down_box DOWN_BOX
+            xywh {10 255 220 20} down_box DOWN_BOX
             code0 {o->value(config.cfg.IgnoreProgramChange);}
           }
         }
diff --git a/src/UI/EnvelopeUI.fl b/src/UI/EnvelopeUI.fl
index fd25ca01..7df1b800 100644
--- a/src/UI/EnvelopeUI.fl
+++ b/src/UI/EnvelopeUI.fl
@@ -312,7 +312,7 @@ sustaincounter->maximum(env->Penvpoints-2);}
         xywh {200 155 80 20} box THIN_UP_BOX labelsize 11
         code0 {if (env->Pfreemode==0) o->hide();}
       }
-      Fl_Light_Button freemodebutton {
+      Fl_Check_Button freemodebutton {
         label FreeMode
         callback {reinit();
 
diff --git a/src/UI/OscilGenUI.fl b/src/UI/OscilGenUI.fl
index a3bc5d63..96bd15df 100644
--- a/src/UI/OscilGenUI.fl
+++ b/src/UI/OscilGenUI.fl
@@ -551,7 +551,7 @@ if ((oscil->Pcurrentbasefunc==0)||(oscil->Pcurrentbasefunc==127)) basefuncmodula
             callback {oscil->Pbasefuncmodulation=(int) o->value();
 basefuncdisplaygroup->redraw();
 redrawoscil();}
-            tooltip {Base function modulation} xywh {599 281 50 15} down_box BORDER_BOX labelsize 10 textsize 10
+            tooltip {Base function modulation} xywh {601 281 50 15} down_box BORDER_BOX labelsize 10 textsize 10
           } {
             MenuItem {} {
               label None
@@ -691,14 +691,14 @@ redrawoscil();}
         xywh {670 505 55 15} box THIN_UP_BOX labelfont 1 labelsize 11
       }
       Fl_Group {} {
-        xywh {135 308 150 30} box UP_FRAME
+        xywh {136 308 150 30} box UP_FRAME
       } {
         Fl_Choice wshbutton {
           label {Wsh.}
           callback {oscil->Pwaveshapingfunction=(int) o->value();
 basefuncdisplaygroup->redraw();
 redrawoscil();} open
-          tooltip {Waveshaping function} xywh {165 313 55 20} down_box BORDER_BOX labelsize 10 textsize 10
+          tooltip {Waveshaping function} xywh {166 313 55 20} down_box BORDER_BOX labelsize 10 textsize 10
         } {
           MenuItem {} {
             label None
@@ -765,26 +765,26 @@ redrawoscil();} open
           callback {oscil->Pwaveshaping=(int)o->value()+64;
 wsparval->value(oscil->Pwaveshaping-64);
 redrawoscil();}
-          tooltip {Waveshaping Parameter} xywh {260 313 20 20} minimum -64 maximum 63 step 1
+          tooltip {Waveshaping Parameter} xywh {261 313 20 20} minimum -64 maximum 63 step 1
           class WidgetPDial
         }
         Fl_Value_Output wsparval {
-          xywh {228 316 25 15} labelsize 12 minimum -63 maximum 63 step 1
+          xywh {229 316 25 15} labelsize 12 minimum -63 maximum 63 step 1
         }
       }
-      Fl_Light_Button autoclearbutton {
+      Fl_Check_Button autoclearbutton {
         label {Clr.}
-        tooltip {Auto clear when using the oscillator as base function} xywh {95 313 35 20} box THIN_UP_BOX value 1 labelfont 1 labelsize 10
+        tooltip {Auto clear when using the oscillator as base function} xywh {94 313 38 20} box THIN_UP_BOX value 1 labelfont 1 labelsize 10
       }
       Fl_Group {} {
-        xywh {285 308 155 30} box UP_FRAME
+        xywh {287 308 155 30} box UP_FRAME
       } {
         Fl_Choice fltbutton {
           label Filter
           callback {oscil->Pfiltertype=(int) o->value();
 
 redrawoscil();}
-          tooltip {Oscillator's filter type} xywh {315 313 50 20} down_box BORDER_BOX labelsize 10 textsize 10
+          tooltip {Oscillator's filter type} xywh {317 313 50 20} down_box BORDER_BOX labelsize 10 textsize 10
         } {
           MenuItem {} {
             label None
@@ -847,7 +847,7 @@ redrawoscil();}
           callback {oscil->Pfilterpar1=(int)o->value();
 
 redrawoscil();}
-          tooltip {Oscillator's filter parameter1} xywh {367 313 20 20} maximum 127 step 1
+          tooltip {Oscillator's filter parameter1} xywh {369 313 20 20} maximum 127 step 1
           class WidgetPDial
         }
         Fl_Check_Button filterpref {
@@ -855,24 +855,24 @@ redrawoscil();}
           callback {oscil->Pfilterbeforews=(int)o->value();
 
 redrawoscil();}
-          tooltip {Apply the filter before the waveshaping} xywh {415 313 20 20} down_box DOWN_BOX labelsize 10 align 24
+          tooltip {Apply the filter before the waveshaping} xywh {417 313 20 20} down_box DOWN_BOX labelsize 10 align 24
         }
         Fl_Dial filtervalue2 {
           callback {oscil->Pfilterpar2=(int)o->value();
 
 redrawoscil();}
-          tooltip {Oscillator's filter parameter2} xywh {392 313 20 20} maximum 127 step 1
+          tooltip {Oscillator's filter parameter2} xywh {394 313 20 20} maximum 127 step 1
           class WidgetPDial
         }
       }
       Fl_Group {} {
-        xywh {590 308 135 30} box UP_FRAME
+        xywh {594 308 135 30} box UP_FRAME
       } {
         Fl_Choice sabutton {
           label {Sp.adj.}
           callback {oscil->Psatype=(int) o->value();
 redrawoscil();}
-          tooltip {Oscillator's spectrum adjust} xywh {630 313 60 20} down_box BORDER_BOX labelsize 10 textsize 10
+          tooltip {Oscillator's spectrum adjust} xywh {635 313 60 20} down_box BORDER_BOX labelsize 10 textsize 10
         } {
           MenuItem {} {
             label None
@@ -894,7 +894,7 @@ redrawoscil();}
         Fl_Dial sadjpar {
           callback {oscil->Psapar=(int)o->value();
 redrawoscil();}
-          tooltip {Oscillator's spectrum adjust parameter} xywh {695 313 20 20} maximum 127 step 1
+          tooltip {Oscillator's spectrum adjust parameter} xywh {702 313 20 20} maximum 127 step 1
           class WidgetPDial
         }
       }
@@ -988,14 +988,14 @@ redrawoscil();}
         }
       }
       Fl_Group {} {
-        xywh {440 308 150 30} box UP_FRAME
+        xywh {443 308 150 30} box UP_FRAME
       } {
         Fl_Choice modtype {
           label {Mod.}
           callback {oscil->Pmodulation=(int) o->value();
 
 redrawoscil();}
-          tooltip modulation xywh {470 315 50 15} down_box BORDER_BOX labelsize 10 textsize 10
+          tooltip modulation xywh {476 315 50 15} down_box BORDER_BOX labelsize 10 textsize 10
         } {
           MenuItem {} {
             label None
@@ -1018,20 +1018,20 @@ redrawoscil();}
           callback {oscil->Pmodulationpar1=(int)o->value();
 
 redrawoscil();}
-          tooltip {Oscillator's modulation parameter 1} xywh {530 315 15 15} maximum 127 step 1
+          tooltip {Oscillator's modulation parameter 1} xywh {534 315 15 15} maximum 127 step 1
           class WidgetPDial
         }
         Fl_Dial modpar2 {
           callback {oscil->Pmodulationpar2=(int)o->value();
 
 redrawoscil();}
-          tooltip {Oscillator's modulation parameter 2} xywh {550 315 15 15} maximum 127 step 1
+          tooltip {Oscillator's modulation parameter 2} xywh {554 315 15 15} maximum 127 step 1
           class WidgetPDial
         }
         Fl_Dial modpar3 {
           callback {oscil->Pmodulationpar3=(int)o->value();
 redrawoscil();}
-          tooltip {Oscillator's modulation parameter 3} xywh {570 315 15 15} maximum 127 step 1
+          tooltip {Oscillator's modulation parameter 3} xywh {574 315 15 15} maximum 127 step 1
           class WidgetPDial
         }
       }
-- 
2.47.0

